# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rustc/sparc.patch
# Copyright (C) 2023 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- rustc-1.78.0-src/vendor/object-0.32.2/src/write/elf/object.rs.vanilla	2024-05-18 18:52:02.723910857 +0200
+++ rustc-1.78.0-src/vendor/object-0.32.2/src/write/elf/object.rs	2024-05-18 18:53:46.458905632 +0200
@@ -142,6 +142,7 @@
             Architecture::S390x => true,
             Architecture::Sbf => false,
             Architecture::Sharc => true,
+            Architecture::Sparc => true,
             Architecture::Sparc64 => true,
             Architecture::Xtensa => true,
             _ => {
@@ -303,6 +304,7 @@
             (Architecture::S390x, None) => elf::EM_S390,
             (Architecture::Sbf, None) => elf::EM_SBF,
             (Architecture::Sharc, None) => elf::EM_SHARC,
+            (Architecture::Sparc, None) => elf::EM_SPARC,
             (Architecture::Sparc64, None) => elf::EM_SPARCV9,
             (Architecture::Xtensa, None) => elf::EM_XTENSA,
             _ => {
--- rustc-1.78.0-src/vendor/object-0.32.2/src/common.rs.vanilla	2024-05-18 19:02:28.062879358 +0200
+++ rustc-1.78.0-src/vendor/object-0.32.2/src/common.rs	2024-05-18 19:03:31.611876157 +0200
@@ -27,6 +27,7 @@
     S390x,
     Sbf,
     Sharc,
+    Sparc,
     Sparc64,
     Wasm32,
     Wasm64,
@@ -70,6 +71,7 @@
             Architecture::S390x => Some(AddressSize::U64),
             Architecture::Sbf => Some(AddressSize::U64),
             Architecture::Sharc => Some(AddressSize::U32),
+            Architecture::Sparc => Some(AddressSize::U32),
             Architecture::Sparc64 => Some(AddressSize::U64),
             Architecture::Wasm32 => Some(AddressSize::U32),
             Architecture::Wasm64 => Some(AddressSize::U64),
